xcp_d/utils/plot.py

```diff
@@ -1000,3 +1001,169 @@ def _get_tr(img):
+        template_file = template_file[0]
+    template = str(template_file)
+    output_path = os.path.abspath(output_path)
+    plott.plot_stat_map(filename,
+                        bg_img=template,
+                        display_mode='mosaic',
+                        cut_coords=8,
+                        output_file=output_path)
+    return output_path
+
+
+def surf_data_from_cifti(data, axis, surf_name):
+    """From https://neurostars.org/t/separate-cifti-by-structure-in-python/17301/2.
+
+    https://nbviewer.org/github/neurohackademy/nh2020-curriculum/blob/master/we-nibabel-markiewicz/NiBabel.ipynb
+    """
+    assert isinstance(axis, nb.cifti2.BrainModelAxis)
+    for name, data_indices, model in axis.iter_structures():
+        # Iterates over volumetric and surface structures
+        if name == surf_name:  # Just looking for a surface
+            data = data.T[data_indices]
+            # Assume brainmodels axis is last, move it to front
+            vtx_indices = model.vertex
+            # Generally 1-N, except medial wall vertices
+            surf_data = np.zeros((vtx_indices.max() + 1,) + data.shape[1:], dtype=data.dtype)
+            surf_data[vtx_indices] = data
+            return surf_data
+
+    raise ValueError(f"No structure named {surf_name}")
+
+
+def plot_alff_reho_surface(output_path, filename, bold_file):
+    """
+    Plot ReHo and ALFF for ciftis on surface.
+
+    Parameters
+    ----------
+    output_path :
+        path to save svg plot
+    filename :
+        surface file
+    bold_file :
+        original input bold file
+
+    Returns
+    ----------
+    output_path:
+        path to svg plot
+
+    """
+    import os
+
+    import matplotlib.pyplot as plt
+    import nibabel as nb
+    from bids.layout import parse_file_entities
+    from nilearn import plotting as plott
+    from templateflow.api import get as get_template
+
+    from xcp_d.utils.plot import surf_data_from_cifti
+
+    density = parse_file_entities(bold_file).get("den", "32k")
+    if density == "91k":
+        density = "32k"
+    rh = str(
+        get_template(template='fsLR', hemi="R", density="32k",
+                     suffix="midthickness", extension=".surf.gii")
+    )
+    lh = str(
+        get_template(template='fsLR', hemi="L", density="32k",
+                     suffix="midthickness", extension=".surf.gii")
+    )
+
+    cifti = nb.load(filename)
+    cifti_data = cifti.get_fdata()
+    cifti_axes = [cifti.header.get_axis(i) for i in range(cifti.ndim)]
+
+    fig, axes = plt.subplots(figsize=(16, 16), ncols=2, nrows=2, subplot_kw={'projection': '3d'})
+    output_path = os.path.abspath(output_path)
+    lh_surf_data = surf_data_from_cifti(
+        cifti_data,
+        cifti_axes[1],
+        'CIFTI_STRUCTURE_CORTEX_LEFT',
+    )
+    rh_surf_data = surf_data_from_cifti(
+        cifti_data,
+        cifti_axes[1],
+        'CIFTI_STRUCTURE_CORTEX_RIGHT',
+    )
+    plott.plot_surf_stat_map(
```

Comment:
Optimally, it would be nice to have a colorer, but I'm not sure how to do that without making everything look terrible. Maybe it can be an open issue. In the meantime, we should at least ensure that the minimum and maximum values are the same across the four subplots. Try extracting the minimum and maximum values across both hemispheres' surface data and then using those to set vmin and vmax in all of the subplots.